pubpub/pubpub
# update: Pub grid refactor, full-bleed support, mobile discussions

This PR refactors the Pub contents to use a CSS grid-based layout in order to support full-bleed content in the Pub document. By stretching a grid across the entire viewport (with 100% width and no margin or padding) and defining the pub content area, gutter, and discussions column entirely within CSS grid, we can trivially make content elements stretch across the viewport by assigning them `grid-column: 1 / -1`. In doing this refactor, I also had to make a change that simplifies the layout calculations for pub-side discussion, and I took this chance to make them work in mobile views (by hiding the commenters' names when closed and forcing expanded discussions inline) as well.

I've enabled full-bleed content by way of a new "breakout" option on images, videos, and iframe embeds. When enabled, the left/right/center/full alignment options for these elements will calculate positions relative to the edges of the viewport, rather than the content.

_Notes:_
- Having multiple grids on the page means we'll have to be careful about keeping them in sync. Until CSS subgrids become widely available, this will be hard to avoid, since elements with `display: grid` can only apply their grid layout to their direct children.
- We should take the time to disable the "breakout" option in the somewhat uncommon case when the content is nested deeper in the document, such as in a table.

_Test plan:_
- Ensure that the "breakout" option works for images, videos, and iframe embeds
- Make sure that display and positioning for discussions still works, including the new mobile view.

_Full-bleed content in a pub:_
<img width="1552" alt="Screen Shot 2019-06-17 at 1 49 22 PM" src="https://user-images.githubusercontent.com/2208769/59629599-d6249b80-9108-11e9-99e0-719b897f9f5a.png">

_A comment in mobile view:_
<img width="300" src="https://user-images.githubusercontent.com/2208769/59629840-5b0fb500-9109-11e9-8153-69cf68fe2aa8.png"/>
